Frontend/src/pages/registration_administration/components/RegistrationAdministrationList.jsx

```diff
@@ -8,6 +8,63 @@ import LoadingMessage from '../../../ui/loadingMessage'
 import styles from './list.module.scss'
 import RegistrationActions from './RegistrationActions'
 
+const reducer = (state, action) => {
+  const { type, attendee } = action
+  switch (type) {
+    case 'add-waiting':
+      return {
+        waiting: [...state.waiting, attendee],
```

Comment:
Nice, looks good overall. Generally reducers are meant to handle complicated state (like we have here) and ensure that someone using the dispatch can't mess up the state.
A potential issue here is that someone could call the dispatch with `add-_` when the attendee is already added. It would be best to check whether the attendee is already in the array, and if so return the _original_ `state` (as opposed to a new copy of the old state, to avoid triggering unnecessary re-renders - though that's probably not significant at this scale).
Another potential issue is that there's nothing stopping someone (future coder, not user) from using the dispatch to add someone from one section (say deleted) into the waiting array. I'm not sure how best to address this (maybe the state should just be a list of selected registrations, and then the 3 sublists are calculated using that and the registrations data? or the selected state should be merged somehow with the registrations state? would have to sit down and think about it some more), but it's probably fine to ignore this for now and revisit it later (but maybe leave a quick comment?).

Reply:
I added a check if the attendee is already in the state, and added a comment about potentially merging states